test/remote/gateways/remote_payeezy_test.rb

```diff
@@ -156,6 +156,34 @@ def test_failed_purchase_with_insufficient_funds
     assert_match(/Insufficient Funds/, response.message)
   end
 
+  def test_successful_purchase_with_three_ds_data
+    @options[:three_d_secure] = {
+      version: '1',
+      eci: '05',
+      cavv: '3q2+78r+ur7erb7vyv66vv////8=',
+      acs_transaction_id: '6546464645623455665165+qe-jmhabcdefg'
+    }
+    assert response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_match(/Transaction Normal/, response.message)
+    assert_equal '100', response.params['bank_resp_code']
+    assert_equal nil, response.error_code
+    assert_success response
+  end
+
+  def test_authorize_and_capture_three_ds_data
+    @options[:three_d_secure] = {
+      version: '1',
+      eci: '05',
+      cavv: '3q2+78r+ur7erb7vyv66vv////8=',
+      acs_transaction_id: '6546464645623455665165+qe-jmhabcdefg'
+    }
+    assert auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+    assert auth.authorization
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+  end
+
```

Comment:
❓ Is there some failed transaction cases?

Reply:
The sandbox functionality does not validate the fields, so adding tests to this would not be useful. 
Let me know if you think otherwise.
Thanks!